src/lists.py

```diff
@@ -125,4 +145,12 @@ def sort(x: DLList[S]) -> None:
     >>> print(x)
     [1, 3, 4, 5, 6, 12]
     """
-    ...
+    def inner(x) -> bool:
+        link, is_sorted = x.head.next, True
+        while link.next.val is not None:
```

Comment:
Why do you care if values are `None`? True, usually `None` doesn't compare with `<`, but I can make a class that can handle comparisons with `None`, and then your sort function will break if you ignore `None`.